File: rb/features.rb
# Virenmonitoring SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module VirenmonitoringFeatures
  def self.make_feature(name)
    case name
    when "base"
      VirenmonitoringBaseFeature.new
    when "test"
      VirenmonitoringTestFeature.new
    else
      VirenmonitoringBaseFeature.new
    end
  end
end

File: py/virenmonitoring_sdk/features.py
# Virenmonitoring SDK feature factory

from virenmonitoring_sdk.feature.base_feature import VirenmonitoringBaseFeature
from virenmonitoring_sdk.feature.test_feature import VirenmonitoringTestFeature


def _make_feature(name):
    features = {
        "base": lambda: VirenmonitoringBaseFeature(),
        "test": lambda: VirenmonitoringTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
